pkg/types/powervs/validation/platform.go

```diff
@@ -44,6 +50,68 @@ func ValidatePlatform(p *powervs.Platform, fldPath *field.Path) field.ErrorList
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("ServiceInstanceGUID"), p.ServiceInstanceGUID, "ServiceInstanceGUID must be a valid UUID"))
 		}
 	}
+	if p.ServiceEndpoints != nil {
+		allErrs = append(allErrs, validateServiceEndpoints(p.ServiceEndpoints, fldPath.Child("serviceEndpoints"))...)
+	}
+
+	return allErrs
+}
+
+// validateServiceEndpoints checks that the specified ServiceEndpoints are valid.
+func validateServiceEndpoints(endpoints []configv1.PowerVSServiceEndpoint, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	knownEndpoints := sets.New[string]()
+	for index, endpoint := range endpoints {
+		fldp := fldPath.Index(index)
+		if knownEndpoints.Has(endpoint.Name) {
+			allErrs = append(allErrs, field.Duplicate(fldp.Child("name"), endpoint.Name))
+		}
+		knownEndpoints.Insert(endpoint.Name)
 
+		if err := validateServiceURL(endpoint.URL); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldp.Child("url"), endpoint.URL, err.Error()))
+		}
+	}
 	return allErrs
 }
+
+// schemeRE is used to check whether a string starts with a scheme (URI format).
+var schemeRE = regexp.MustCompile("^([^:]+)://")
+
+// versionPath is the regexp for a trailing API version in URL path ('/v1', '/v22/', etc.)
+var versionPath = regexp.MustCompile(`(/v\d+[/]{0,1})$`)
+
+// validateServiceURL checks that a string meets certain URI expectations.
+func validateServiceURL(uri string) error {
+	endpoint := uri
+	httpsScheme := "https"
```

Comment:
In case you have to change something else:
```suggestion
	const httpsScheme = "https"
```